### PR TITLE
Charlesmchen/sender names3

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 1]
```diff
@@ -4923,7 +4923,7 @@ - (void)reloadViewItems
                             || viewItem.shouldShowDate);
                 }
                 if (shouldShowSenderName) {
-                    senderName = [self.contactsManager displayNameForPhoneIdentifier:incomingSenderId];
+                    senderName = [self.contactsManager contactOrProfileNameForPhoneIdentifier:incomingSenderId];
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Is this styled appropriately? e.g. we generally want to use the attributed flavor to make sure profile names are distinctly recognizable from something fetched from our own contacts.

Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: I was wondering the same thing. I'll add a note to discuss with myles.

Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Talked with Myles.  Design isn't final, but I've made some changes per him.

File: SignalMessaging/contacts/OWSContactsManager.m [hunk 2]
```diff
@@ -738,37 +738,49 @@ - (NSAttributedString *)attributedContactOrProfileNameForPhoneIdentifier:(NSStri
...
-        NSRange recipientIdRange = [numberAndProfileName rangeOfString:recipientId];
-        NSMutableAttributedString *attributedString =
-            [[NSMutableAttributedString alloc] initWithString:numberAndProfileName
-                                                   attributes:@{ NSFontAttributeName : secondaryFont }];
-        [attributedString addAttribute:NSFontAttributeName value:primaryFont range:recipientIdRange];
-
-        return [attributedString copy];
+        NSAttributedString *result =
+            [[NSAttributedString alloc] initWithString:recipientId attributes:primaryAttributes];
+        result = [result rtlSafeAppend:[[NSAttributedString alloc] initWithString:@" "] referenceView:[UIView new]];
```
Comment (charlesmchen-signal) [inline SignalMessaging/contacts/OWSContactsManager.m]: IMO it's safer to use "RTL safe append" than to have translators juggle complicated and hard-to-read format string. WDYT?  If you approve, I'd like to remove all usage of `PROFILE_NAME_AND_PHONE_NUMBER_LABEL_FORMAT`.

**Comment by charlesmchen-signal** [comment]
PTAL @michaelkirk

**Pull request merged**